### PR TITLE
Provide better error message for --compare_path without testing dependencies

### PR DESCRIPTION
**Pull request by biosafetylvl5**
* [x] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
* [x] NO REQUIRED ***unit tests*** (explain why not required)
* [x] Required ***integration tests*** added and pass for new/modified functionality
* [x] NO REQUIRED ***documentation*** (explain why not required)
* [x] Required ***release notes*** added for new/modified functionality
* [x] NO REQUIRED ***updates to other repos*** (explain why not required)

# Related Issues
fixes NRLMMD-GEOIPS/geoips#451

# Testing Instructions
* Enter a `bash` environment in the Dockerfile, remove `pixelmatch`, run `tests/integration_tests/base_install.sh` and then `tests/integration_tests/base_test.sh`. `cat /output/logs/20240408/20240408.223821_geoips_base/test_all_geoips_base.log_amsr2_ocean.tc.windspeed.imagery_clean.sh.log` to see error message:

```bash
Processing complete! Checking outputs...


Traceback (most recent call last):
  File "/app/geoips_packages/geoips/geoips/plugins/modules/output_checkers/image.py", line 123, in outputs_match
    from pixelmatch.contrib.PIL import pixelmatch
ModuleNotFoundError: No module named 'pixelmatch'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/geoips_user/.local/bin/run_procflow", line 8, in <module>
    sys.exit(main())
  File "/app/geoips_packages/geoips/geoips/commandline/run_procflow.py", line 69, in main
    RETVAL = PROCFLOW(COMMAND_LINE_ARGS["filenames"], COMMAND_LINE_ARGS)
  File "/app/geoips_packages/geoips/geoips/plugins/modules/procflows/single_source.py", line 1917, in call
    retval += output_checker(
  File "/app/geoips_packages/geoips/geoips/plugins/modules/output_checkers/image.py", line 230, in call
    retval = plugin.compare_outputs(
  File "/app/geoips_packages/geoips/geoips/interfaces/module_based/output_checkers.py", line 849, in compare_outputs
    ) = self.perform_comparisons(
  File "/app/geoips_packages/geoips/geoips/interfaces/module_based/output_checkers.py", line 776, in perform_comparisons
    goodcomps, badcomps, compare_strings = test_product_func(
  File "/app/geoips_packages/geoips/geoips/interfaces/module_based/output_checkers.py", line 965, in test_products
    if self.module.outputs_match(
  File "/app/geoips_packages/geoips/geoips/plugins/modules/output_checkers/image.py", line 125, in outputs_match
    raise ImportError(
ImportError: Comparing images using GeoIPS requires a python library called pixelmatch (link: https://pypi.org/project/pixelmatch/).You can install it directly or install the test extra eg. pip install geoips[test]
```

# Summary
* Added a clear and informative error message when `--compare_path` is used without the pixelmatch dependency installed.
* Checked for other testing functionalities in `run_procflow` and added appropriate error messages for missing dependencies. See #451 for details.

# Output
* Command line output demonstrating the new error message when `--compare_path` is used without installing pixelmatch.
* No imagery output is required as the changes are related to error handling.

### CODE REVIEW

File: geoips/plugins/modules/output_checkers/image.py
Comment (evrose54) [inline geoips/plugins/modules/output_checkers/image.py:122]: We should wait on merging this PR until #438 is merged. Then, we should replace the optional import below with the new ``import_optional_dependencies`` functionality that has been created in #438.

Comment (jsolbrig) [reply geoips/plugins/modules/output_checkers/image.py:122]: @biosafetylvl5 This comment is ready to be addressed.

Comment (biosafetylvl5) [reply geoips/plugins/modules/output_checkers/image.py:122]: @evrose54 @jsolbrig The error message below has more details, and would be more useful to users. What benefit do we get by replacing with the `import_optional_dependencies` function? I'm so fine with doing so, but from what I see, it seems worse for the user. It also appears to fail to raise the error exception with traceback?

Let me know what I'm missing here 😵‍💫

Comment (jsolbrig) [reply geoips/plugins/modules/output_checkers/image.py:122]: In this case I agree with you. Since you've implemented a better error message and this is a special case, I think we should keep this.

The history behind `import_optional_dependencies` is that there were dozens of these optional imports scattered all over the code, many with useless error messages. Maybe import_optional_dependencies wasn't the right solution, but it helped at least add something useful to those import errors to tell people what to do.

That said, the current error message still isn't very helpful and should be improved. We should think about how to improve the error message and how to get rid of as many optional imports as possible.

Comment (biosafetylvl5) [reply geoips/plugins/modules/output_checkers/image.py:122]: Perhaps we can implement package specific messages, so it'd - for example - be applicable here, but still centralized. For now, I think this PR is gtg.

**Review by jsolbrig — APPROVED** [review]

**Review by mindyls — APPROVED** [review]
Looks good to me.

**Pull request merged**
